[2025-05-20T20:47:20.275+0000] {taskinstance.py:1157} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: yellow_taxi_batch_pipeline.transform_taxi_data manual__2025-05-20T20:45:38.757554+00:00 [queued]>
[2025-05-20T20:47:20.322+0000] {taskinstance.py:1157} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: yellow_taxi_batch_pipeline.transform_taxi_data manual__2025-05-20T20:45:38.757554+00:00 [queued]>
[2025-05-20T20:47:20.323+0000] {taskinstance.py:1359} INFO - Starting attempt 1 of 2
[2025-05-20T20:47:20.384+0000] {taskinstance.py:1380} INFO - Executing <Task(BashOperator): transform_taxi_data> on 2025-05-20 20:45:38.757554+00:00
[2025-05-20T20:47:20.411+0000] {standard_task_runner.py:57} INFO - Started process 802 to run task
[2025-05-20T20:47:20.418+0000] {standard_task_runner.py:84} INFO - Running: ['airflow', 'tasks', 'run', 'yellow_taxi_batch_pipeline', 'transform_taxi_data', 'manual__2025-05-20T20:45:38.757554+00:00', '--job-id', '44', '--raw', '--subdir', 'DAGS_FOLDER/taxi_batch_dag.py', '--cfg-path', '/tmp/tmpyprhy2sf']
[2025-05-20T20:47:20.427+0000] {standard_task_runner.py:85} INFO - Job 44: Subtask transform_taxi_data
[2025-05-20T20:47:20.484+0000] {logging_mixin.py:151} WARNING - /home/airflow/.local/lib/python3.8/site-packages/airflow/settings.py:195 DeprecationWarning: The sql_alchemy_conn option in [core] has been moved to the sql_alchemy_conn option in [database] - the old setting has been used, but please update your config.
[2025-05-20T20:47:20.628+0000] {task_command.py:415} INFO - Running <TaskInstance: yellow_taxi_batch_pipeline.transform_taxi_data manual__2025-05-20T20:45:38.757554+00:00 [running]> on host 54a986196d48
[2025-05-20T20:47:20.951+0000] {taskinstance.py:1660} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='airflow' AIRFLOW_CTX_DAG_ID='yellow_taxi_batch_pipeline' AIRFLOW_CTX_TASK_ID='transform_taxi_data' AIRFLOW_CTX_EXECUTION_DATE='2025-05-20T20:45:38.757554+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-05-20T20:45:38.757554+00:00'
[2025-05-20T20:47:20.961+0000] {subprocess.py:63} INFO - Tmp dir root location: /tmp
[2025-05-20T20:47:20.964+0000] {subprocess.py:75} INFO - Running command: ['/bin/bash', '-c', '\n    spark-submit         --master spark://spark-master:7077         --jars /opt/airflow/dags/jars/***ql-42.5.0.jar         --packages org.apache.hadoop:hadoop-aws:3.3.1         /opt/airflow/dags/scripts/taxi_transform.py\n    ']
[2025-05-20T20:47:20.994+0000] {subprocess.py:86} INFO - Output:
[2025-05-20T20:47:21.027+0000] {subprocess.py:93} INFO - /opt/spark/bin/load-spark-env.sh: line 68: ps: command not found
[2025-05-20T20:47:28.853+0000] {subprocess.py:93} INFO - :: loading settings :: url = jar:file:/opt/spark/jars/ivy-2.5.0.jar!/org/apache/ivy/core/settings/ivysettings.xml
[2025-05-20T20:47:29.202+0000] {subprocess.py:93} INFO - Ivy Default Cache set to: /home/airflow/.ivy2/cache
[2025-05-20T20:47:29.203+0000] {subprocess.py:93} INFO - The jars for the packages stored in: /home/airflow/.ivy2/jars
[2025-05-20T20:47:29.224+0000] {subprocess.py:93} INFO - org.apache.hadoop#hadoop-aws added as a dependency
[2025-05-20T20:47:29.236+0000] {subprocess.py:93} INFO - :: resolving dependencies :: org.apache.spark#spark-submit-parent-6f424da7-be95-4a51-8b34-6711b7ae698d;1.0
[2025-05-20T20:47:29.239+0000] {subprocess.py:93} INFO - 	confs: [default]
[2025-05-20T20:47:30.309+0000] {subprocess.py:93} INFO - 	found org.apache.hadoop#hadoop-aws;3.3.1 in central
[2025-05-20T20:47:30.525+0000] {subprocess.py:93} INFO - 	found com.amazonaws#aws-java-sdk-bundle;1.11.901 in central
[2025-05-20T20:47:30.796+0000] {subprocess.py:93} INFO - 	found org.wildfly.openssl#wildfly-openssl;1.0.7.Final in central
[2025-05-20T20:47:30.938+0000] {subprocess.py:93} INFO - :: resolution report :: resolve 1655ms :: artifacts dl 49ms
[2025-05-20T20:47:30.960+0000] {subprocess.py:93} INFO - 	:: modules in use:
[2025-05-20T20:47:30.970+0000] {subprocess.py:93} INFO - 	com.amazonaws#aws-java-sdk-bundle;1.11.901 from central in [default]
[2025-05-20T20:47:30.974+0000] {subprocess.py:93} INFO - 	org.apache.hadoop#hadoop-aws;3.3.1 from central in [default]
[2025-05-20T20:47:31.002+0000] {subprocess.py:93} INFO - 	org.wildfly.openssl#wildfly-openssl;1.0.7.Final from central in [default]
[2025-05-20T20:47:31.006+0000] {subprocess.py:93} INFO - 	---------------------------------------------------------------------
[2025-05-20T20:47:31.008+0000] {subprocess.py:93} INFO - 	|                  |            modules            ||   artifacts   |
[2025-05-20T20:47:31.010+0000] {subprocess.py:93} INFO - 	|       conf       | number| search|dwnlded|evicted|| number|dwnlded|
[2025-05-20T20:47:31.012+0000] {subprocess.py:93} INFO - 	---------------------------------------------------------------------
[2025-05-20T20:47:31.012+0000] {subprocess.py:93} INFO - 	|      default     |   3   |   0   |   0   |   0   ||   3   |   0   |
[2025-05-20T20:47:31.013+0000] {subprocess.py:93} INFO - 	---------------------------------------------------------------------
[2025-05-20T20:47:31.014+0000] {subprocess.py:93} INFO - :: retrieving :: org.apache.spark#spark-submit-parent-6f424da7-be95-4a51-8b34-6711b7ae698d
[2025-05-20T20:47:31.016+0000] {subprocess.py:93} INFO - 	confs: [default]
[2025-05-20T20:47:31.024+0000] {subprocess.py:93} INFO - 	0 artifacts copied, 3 already retrieved (0kB/25ms)
[2025-05-20T20:47:32.642+0000] {subprocess.py:93} INFO - 25/05/20 20:47:32 WARN NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
[2025-05-20T20:47:33.816+0000] {subprocess.py:93} INFO - 25/05/20 20:47:33 WARN DependencyUtils: Local jar /opt/airflow/dags/jars/***ql-42.5.0.jar does not exist, skipping.
[2025-05-20T20:47:39.126+0000] {subprocess.py:93} INFO - 25/05/20 20:47:39 INFO SparkContext: Running Spark version 3.3.1
[2025-05-20T20:47:39.224+0000] {subprocess.py:93} INFO - 25/05/20 20:47:39 INFO ResourceUtils: ==============================================================
[2025-05-20T20:47:39.232+0000] {subprocess.py:93} INFO - 25/05/20 20:47:39 INFO ResourceUtils: No custom resources configured for spark.driver.
[2025-05-20T20:47:39.235+0000] {subprocess.py:93} INFO - 25/05/20 20:47:39 INFO ResourceUtils: ==============================================================
[2025-05-20T20:47:39.237+0000] {subprocess.py:93} INFO - 25/05/20 20:47:39 INFO SparkContext: Submitted application: YellowTaxiTransform
[2025-05-20T20:47:39.336+0000] {subprocess.py:93} INFO - 25/05/20 20:47:39 INFO ResourceProfile: Default ResourceProfile created, executor resources: Map(cores -> name: cores, amount: 1, script: , vendor: , memory -> name: memory, amount: 1024, script: , vendor: , offHeap -> name: offHeap, amount: 0, script: , vendor: ), task resources: Map(cpus -> name: cpus, amount: 1.0)
[2025-05-20T20:47:39.515+0000] {subprocess.py:93} INFO - 25/05/20 20:47:39 INFO ResourceProfile: Limiting resource is cpu
[2025-05-20T20:47:39.518+0000] {subprocess.py:93} INFO - 25/05/20 20:47:39 INFO ResourceProfileManager: Added ResourceProfile id: 0
[2025-05-20T20:47:39.753+0000] {subprocess.py:93} INFO - 25/05/20 20:47:39 INFO SecurityManager: Changing view acls to: airflow
[2025-05-20T20:47:39.762+0000] {subprocess.py:93} INFO - 25/05/20 20:47:39 INFO SecurityManager: Changing modify acls to: airflow
[2025-05-20T20:47:39.763+0000] {subprocess.py:93} INFO - 25/05/20 20:47:39 INFO SecurityManager: Changing view acls groups to:
[2025-05-20T20:47:39.763+0000] {subprocess.py:93} INFO - 25/05/20 20:47:39 INFO SecurityManager: Changing modify acls groups to:
[2025-05-20T20:47:39.764+0000] {subprocess.py:93} INFO - 25/05/20 20:47:39 INFO SecurityManager: SecurityManager: authentication disabled; ui acls disabled; users  with view permissions: Set(airflow); groups with view permissions: Set(); users  with modify permissions: Set(airflow); groups with modify permissions: Set()
[2025-05-20T20:47:41.778+0000] {subprocess.py:93} INFO - 25/05/20 20:47:41 INFO Utils: Successfully started service 'sparkDriver' on port 46005.
[2025-05-20T20:47:42.027+0000] {subprocess.py:93} INFO - 25/05/20 20:47:42 INFO SparkEnv: Registering MapOutputTracker
[2025-05-20T20:47:42.176+0000] {subprocess.py:93} INFO - 25/05/20 20:47:42 INFO SparkEnv: Registering BlockManagerMaster
[2025-05-20T20:47:42.304+0000] {subprocess.py:93} INFO - 25/05/20 20:47:42 INFO BlockManagerMasterEndpoint: Using org.apache.spark.storage.DefaultTopologyMapper for getting topology information
[2025-05-20T20:47:42.305+0000] {subprocess.py:93} INFO - 25/05/20 20:47:42 INFO BlockManagerMasterEndpoint: BlockManagerMasterEndpoint up
[2025-05-20T20:47:42.323+0000] {subprocess.py:93} INFO - 25/05/20 20:47:42 INFO SparkEnv: Registering BlockManagerMasterHeartbeat
[2025-05-20T20:47:42.455+0000] {subprocess.py:93} INFO - 25/05/20 20:47:42 INFO DiskBlockManager: Created local directory at /tmp/blockmgr-3ddc79a0-3722-4a50-8bfd-862851b8cb0d
[2025-05-20T20:47:42.519+0000] {subprocess.py:93} INFO - 25/05/20 20:47:42 INFO MemoryStore: MemoryStore started with capacity 434.4 MiB
[2025-05-20T20:47:42.581+0000] {subprocess.py:93} INFO - 25/05/20 20:47:42 INFO SparkEnv: Registering OutputCommitCoordinator
[2025-05-20T20:47:43.832+0000] {subprocess.py:93} INFO - 25/05/20 20:47:43 INFO Utils: Successfully started service 'SparkUI' on port 4040.
[2025-05-20T20:47:44.083+0000] {subprocess.py:93} INFO - 25/05/20 20:47:44 ERROR SparkContext: Failed to add /opt/spark/jars/***ql-42.5.0.jar to Spark environment
[2025-05-20T20:47:44.085+0000] {subprocess.py:93} INFO - java.io.FileNotFoundException: Jar /opt/spark/jars/***ql-42.5.0.jar not found
[2025-05-20T20:47:44.088+0000] {subprocess.py:93} INFO - 	at org.apache.spark.SparkContext.addLocalJarFile$1(SparkContext.scala:1949)
[2025-05-20T20:47:44.094+0000] {subprocess.py:93} INFO - 	at org.apache.spark.SparkContext.addJar(SparkContext.scala:2004)
[2025-05-20T20:47:44.100+0000] {subprocess.py:93} INFO - 	at org.apache.spark.SparkContext.$anonfun$new$12(SparkContext.scala:507)
[2025-05-20T20:47:44.106+0000] {subprocess.py:93} INFO - 	at org.apache.spark.SparkContext.$anonfun$new$12$adapted(SparkContext.scala:507)
[2025-05-20T20:47:44.108+0000] {subprocess.py:93} INFO - 	at scala.collection.mutable.ResizableArray.foreach(ResizableArray.scala:62)
[2025-05-20T20:47:44.114+0000] {subprocess.py:93} INFO - 	at scala.collection.mutable.ResizableArray.foreach$(ResizableArray.scala:55)
[2025-05-20T20:47:44.118+0000] {subprocess.py:93} INFO - 	at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:49)
[2025-05-20T20:47:44.125+0000] {subprocess.py:93} INFO - 	at org.apache.spark.SparkContext.<init>(SparkContext.scala:507)
[2025-05-20T20:47:44.128+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.java.JavaSparkContext.<init>(JavaSparkContext.scala:58)
[2025-05-20T20:47:44.137+0000] {subprocess.py:93} INFO - 	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
[2025-05-20T20:47:44.146+0000] {subprocess.py:93} INFO - 	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
[2025-05-20T20:47:44.164+0000] {subprocess.py:93} INFO - 	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
[2025-05-20T20:47:44.167+0000] {subprocess.py:93} INFO - 	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
[2025-05-20T20:47:44.170+0000] {subprocess.py:93} INFO - 	at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:247)
[2025-05-20T20:47:44.171+0000] {subprocess.py:93} INFO - 	at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:357)
[2025-05-20T20:47:44.176+0000] {subprocess.py:93} INFO - 	at py4j.Gateway.invoke(Gateway.java:238)
[2025-05-20T20:47:44.204+0000] {subprocess.py:93} INFO - 	at py4j.commands.ConstructorCommand.invokeConstructor(ConstructorCommand.java:80)
[2025-05-20T20:47:44.219+0000] {subprocess.py:93} INFO - 	at py4j.commands.ConstructorCommand.execute(ConstructorCommand.java:69)
[2025-05-20T20:47:44.232+0000] {subprocess.py:93} INFO - 	at py4j.ClientServerConnection.waitForCommands(ClientServerConnection.java:182)
[2025-05-20T20:47:44.254+0000] {subprocess.py:93} INFO - 	at py4j.ClientServerConnection.run(ClientServerConnection.java:106)
[2025-05-20T20:47:44.258+0000] {subprocess.py:93} INFO - 	at java.base/java.lang.Thread.run(Thread.java:829)
[2025-05-20T20:47:44.262+0000] {subprocess.py:93} INFO - 25/05/20 20:47:44 INFO SparkContext: Added file file:///home/airflow/.ivy2/jars/org.apache.hadoop_hadoop-aws-3.3.1.jar at spark://54a986196d48:46005/files/org.apache.hadoop_hadoop-aws-3.3.1.jar with timestamp 1747774059084
[2025-05-20T20:47:44.267+0000] {subprocess.py:93} INFO - 25/05/20 20:47:44 INFO Utils: Copying /home/airflow/.ivy2/jars/org.apache.hadoop_hadoop-aws-3.3.1.jar to /tmp/spark-d60f1fe2-342e-4414-a5fa-8338cb3959b9/userFiles-02014632-6116-469e-b7fd-5170e6258367/org.apache.hadoop_hadoop-aws-3.3.1.jar
[2025-05-20T20:47:44.269+0000] {subprocess.py:93} INFO - 25/05/20 20:47:44 INFO SparkContext: Added file file:///home/airflow/.ivy2/jars/com.amazonaws_aws-java-sdk-bundle-1.11.901.jar at spark://54a986196d48:46005/files/com.amazonaws_aws-java-sdk-bundle-1.11.901.jar with timestamp 1747774059084
[2025-05-20T20:47:44.273+0000] {subprocess.py:93} INFO - 25/05/20 20:47:44 INFO Utils: Copying /home/airflow/.ivy2/jars/com.amazonaws_aws-java-sdk-bundle-1.11.901.jar to /tmp/spark-d60f1fe2-342e-4414-a5fa-8338cb3959b9/userFiles-02014632-6116-469e-b7fd-5170e6258367/com.amazonaws_aws-java-sdk-bundle-1.11.901.jar
[2025-05-20T20:47:48.743+0000] {subprocess.py:93} INFO - 25/05/20 20:47:48 INFO SparkContext: Added file file:///home/airflow/.ivy2/jars/org.wildfly.openssl_wildfly-openssl-1.0.7.Final.jar at spark://54a986196d48:46005/files/org.wildfly.openssl_wildfly-openssl-1.0.7.Final.jar with timestamp 1747774059084
[2025-05-20T20:47:48.744+0000] {subprocess.py:93} INFO - 25/05/20 20:47:48 INFO Utils: Copying /home/airflow/.ivy2/jars/org.wildfly.openssl_wildfly-openssl-1.0.7.Final.jar to /tmp/spark-d60f1fe2-342e-4414-a5fa-8338cb3959b9/userFiles-02014632-6116-469e-b7fd-5170e6258367/org.wildfly.openssl_wildfly-openssl-1.0.7.Final.jar
[2025-05-20T20:47:49.251+0000] {subprocess.py:93} INFO - 25/05/20 20:47:49 INFO StandaloneAppClient$ClientEndpoint: Connecting to master spark://spark-master:7077...
[2025-05-20T20:47:49.524+0000] {subprocess.py:93} INFO - 25/05/20 20:47:49 INFO TransportClientFactory: Successfully created connection to spark-master/172.18.0.3:7077 after 186 ms (0 ms spent in bootstraps)
[2025-05-20T20:47:50.342+0000] {subprocess.py:93} INFO - 25/05/20 20:47:50 INFO StandaloneSchedulerBackend: Connected to Spark cluster with app ID app-20250520204750-0002
[2025-05-20T20:47:50.350+0000] {subprocess.py:93} INFO - 25/05/20 20:47:50 INFO StandaloneAppClient$ClientEndpoint: Executor added: app-20250520204750-0002/0 on worker-20250520204256-172.18.0.7-35877 (172.18.0.7:35877) with 2 core(s)
[2025-05-20T20:47:50.400+0000] {subprocess.py:93} INFO - 25/05/20 20:47:50 INFO StandaloneSchedulerBackend: Granted executor ID app-20250520204750-0002/0 on hostPort 172.18.0.7:35877 with 2 core(s), 1024.0 MiB RAM
[2025-05-20T20:47:50.438+0000] {subprocess.py:93} INFO - 25/05/20 20:47:50 INFO Utils: Successfully started service 'org.apache.spark.network.netty.NettyBlockTransferService' on port 36289.
[2025-05-20T20:47:50.439+0000] {subprocess.py:93} INFO - 25/05/20 20:47:50 INFO NettyBlockTransferService: Server created on 54a986196d48:36289
[2025-05-20T20:47:50.442+0000] {subprocess.py:93} INFO - 25/05/20 20:47:50 INFO BlockManager: Using org.apache.spark.storage.RandomBlockReplicationPolicy for block replication policy
[2025-05-20T20:47:50.474+0000] {subprocess.py:93} INFO - 25/05/20 20:47:50 INFO BlockManagerMaster: Registering BlockManager BlockManagerId(driver, 54a986196d48, 36289, None)
[2025-05-20T20:47:50.484+0000] {subprocess.py:93} INFO - 25/05/20 20:47:50 INFO BlockManagerMasterEndpoint: Registering block manager 54a986196d48:36289 with 434.4 MiB RAM, BlockManagerId(driver, 54a986196d48, 36289, None)
[2025-05-20T20:47:50.499+0000] {subprocess.py:93} INFO - 25/05/20 20:47:50 INFO BlockManagerMaster: Registered BlockManager BlockManagerId(driver, 54a986196d48, 36289, None)
[2025-05-20T20:47:50.502+0000] {subprocess.py:93} INFO - 25/05/20 20:47:50 INFO BlockManager: Initialized BlockManager: BlockManagerId(driver, 54a986196d48, 36289, None)
[2025-05-20T20:47:50.876+0000] {subprocess.py:93} INFO - 25/05/20 20:47:50 INFO StandaloneAppClient$ClientEndpoint: Executor updated: app-20250520204750-0002/0 is now RUNNING
[2025-05-20T20:47:51.589+0000] {subprocess.py:93} INFO - 25/05/20 20:47:51 INFO StandaloneSchedulerBackend: SchedulerBackend is ready for scheduling beginning after reached minRegisteredResourcesRatio: 0.0
[2025-05-20T20:47:53.009+0000] {subprocess.py:93} INFO - 25/05/20 20:47:53 INFO SharedState: Setting hive.metastore.warehouse.dir ('null') to the value of spark.sql.warehouse.dir.
[2025-05-20T20:47:53.030+0000] {subprocess.py:93} INFO - 25/05/20 20:47:53 INFO SharedState: Warehouse path is 'file:/tmp/airflowtmpg8_ugfbw/spark-warehouse'.
[2025-05-20T20:47:56.311+0000] {subprocess.py:93} INFO - 2025-05-20 20:47:56,310 - INFO - Loading Yellow Taxi data from MinIO
[2025-05-20T20:47:57.318+0000] {subprocess.py:93} INFO - 25/05/20 20:47:57 WARN MetricsConfig: Cannot locate configuration: tried hadoop-metrics2-s3a-file-system.properties,hadoop-metrics2.properties
[2025-05-20T20:47:57.399+0000] {subprocess.py:93} INFO - 25/05/20 20:47:57 INFO MetricsSystemImpl: Scheduled Metric snapshot period at 10 second(s).
[2025-05-20T20:47:57.405+0000] {subprocess.py:93} INFO - 25/05/20 20:47:57 INFO MetricsSystemImpl: s3a-file-system metrics system started
[2025-05-20T20:48:04.313+0000] {subprocess.py:93} INFO - 25/05/20 20:48:04 WARN FileStreamSink: Assume no metadata directory. Error while looking for metadata directory in the path: s3a://nyc-taxi-data/yellow_taxi/yellow_tripdata_2022-*.parquet.
[2025-05-20T20:48:04.328+0000] {subprocess.py:93} INFO - java.nio.file.AccessDeniedException: s3a://nyc-taxi-data/yellow_taxi/yellow_tripdata_2022-*.parquet: org.apache.hadoop.fs.s3a.auth.NoAuthWithAWSException: No AWS Credentials provided by TemporaryAWSCredentialsProvider SimpleAWSCredentialsProvider EnvironmentVariableCredentialsProvider IAMInstanceCredentialsProvider : com.amazonaws.SdkClientException: Unable to load AWS credentials from environment variables (AWS_ACCESS_KEY_ID (or AWS_ACCESS_KEY) and AWS_SECRET_KEY (or AWS_SECRET_ACCESS_KEY))
[2025-05-20T20:48:04.334+0000] {subprocess.py:93} INFO - 	at org.apache.hadoop.fs.s3a.S3AUtils.translateException(S3AUtils.java:206)
[2025-05-20T20:48:04.337+0000] {subprocess.py:93} INFO - 	at org.apache.hadoop.fs.s3a.S3AUtils.translateException(S3AUtils.java:170)
[2025-05-20T20:48:04.343+0000] {subprocess.py:93} INFO - 	at org.apache.hadoop.fs.s3a.S3AFileSystem.s3GetFileStatus(S3AFileSystem.java:3351)
[2025-05-20T20:48:04.345+0000] {subprocess.py:93} INFO - 	at org.apache.hadoop.fs.s3a.S3AFileSystem.innerGetFileStatus(S3AFileSystem.java:3185)
[2025-05-20T20:48:04.349+0000] {subprocess.py:93} INFO - 	at org.apache.hadoop.fs.s3a.S3AFileSystem.isDirectory(S3AFileSystem.java:4277)
[2025-05-20T20:48:04.354+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.FileStreamSink$.hasMetadata(FileStreamSink.scala:54)
[2025-05-20T20:48:04.356+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.DataSource.resolveRelation(DataSource.scala:370)
[2025-05-20T20:48:04.363+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.DataFrameReader.loadV1Source(DataFrameReader.scala:228)
[2025-05-20T20:48:04.366+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.DataFrameReader.$anonfun$load$2(DataFrameReader.scala:210)
[2025-05-20T20:48:04.370+0000] {subprocess.py:93} INFO - 	at scala.Option.getOrElse(Option.scala:189)
[2025-05-20T20:48:04.373+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.DataFrameReader.load(DataFrameReader.scala:210)
[2025-05-20T20:48:04.377+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.DataFrameReader.parquet(DataFrameReader.scala:562)
[2025-05-20T20:48:04.394+0000] {subprocess.py:93} INFO - 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[2025-05-20T20:48:04.397+0000] {subprocess.py:93} INFO - 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[2025-05-20T20:48:04.399+0000] {subprocess.py:93} INFO - 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[2025-05-20T20:48:04.401+0000] {subprocess.py:93} INFO - 	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
[2025-05-20T20:48:04.405+0000] {subprocess.py:93} INFO - 	at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)
[2025-05-20T20:48:04.416+0000] {subprocess.py:93} INFO - 	at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:357)
[2025-05-20T20:48:04.418+0000] {subprocess.py:93} INFO - 	at py4j.Gateway.invoke(Gateway.java:282)
[2025-05-20T20:48:04.437+0000] {subprocess.py:93} INFO - 	at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)
[2025-05-20T20:48:04.448+0000] {subprocess.py:93} INFO - 	at py4j.commands.CallCommand.execute(CallCommand.java:79)
[2025-05-20T20:48:04.449+0000] {subprocess.py:93} INFO - 	at py4j.ClientServerConnection.waitForCommands(ClientServerConnection.java:182)
[2025-05-20T20:48:04.450+0000] {subprocess.py:93} INFO - 	at py4j.ClientServerConnection.run(ClientServerConnection.java:106)
[2025-05-20T20:48:04.453+0000] {subprocess.py:93} INFO - 	at java.base/java.lang.Thread.run(Thread.java:829)
[2025-05-20T20:48:04.455+0000] {subprocess.py:93} INFO - Caused by: org.apache.hadoop.fs.s3a.auth.NoAuthWithAWSException: No AWS Credentials provided by TemporaryAWSCredentialsProvider SimpleAWSCredentialsProvider EnvironmentVariableCredentialsProvider IAMInstanceCredentialsProvider : com.amazonaws.SdkClientException: Unable to load AWS credentials from environment variables (AWS_ACCESS_KEY_ID (or AWS_ACCESS_KEY) and AWS_SECRET_KEY (or AWS_SECRET_ACCESS_KEY))
[2025-05-20T20:48:04.457+0000] {subprocess.py:93} INFO - 	at org.apache.hadoop.fs.s3a.AWSCredentialProviderList.getCredentials(AWSCredentialProviderList.java:216)
[2025-05-20T20:48:04.458+0000] {subprocess.py:93} INFO - 	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.getCredentialsFromContext(AmazonHttpClient.java:1257)
[2025-05-20T20:48:04.460+0000] {subprocess.py:93} INFO - 	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.runBeforeRequestHandlers(AmazonHttpClient.java:833)
[2025-05-20T20:48:04.463+0000] {subprocess.py:93} INFO - 	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.doExecute(AmazonHttpClient.java:783)
[2025-05-20T20:48:04.468+0000] {subprocess.py:93} INFO - 	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeWithTimer(AmazonHttpClient.java:770)
[2025-05-20T20:48:04.469+0000] {subprocess.py:93} INFO - 	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.execute(AmazonHttpClient.java:744)
[2025-05-20T20:48:04.470+0000] {subprocess.py:93} INFO - 	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.access$500(AmazonHttpClient.java:704)
[2025-05-20T20:48:04.471+0000] {subprocess.py:93} INFO - 	at com.amazonaws.http.AmazonHttpClient$RequestExecutionBuilderImpl.execute(AmazonHttpClient.java:686)
[2025-05-20T20:48:04.472+0000] {subprocess.py:93} INFO - 	at com.amazonaws.http.AmazonHttpClient.execute(AmazonHttpClient.java:550)
[2025-05-20T20:48:04.472+0000] {subprocess.py:93} INFO - 	at com.amazonaws.http.AmazonHttpClient.execute(AmazonHttpClient.java:530)
[2025-05-20T20:48:04.473+0000] {subprocess.py:93} INFO - 	at com.amazonaws.services.s3.AmazonS3Client.invoke(AmazonS3Client.java:5227)
[2025-05-20T20:48:04.474+0000] {subprocess.py:93} INFO - 	at com.amazonaws.services.s3.AmazonS3Client.getBucketRegionViaHeadRequest(AmazonS3Client.java:6189)
[2025-05-20T20:48:04.475+0000] {subprocess.py:93} INFO - 	at com.amazonaws.services.s3.AmazonS3Client.fetchRegionFromCache(AmazonS3Client.java:6162)
[2025-05-20T20:48:04.475+0000] {subprocess.py:93} INFO - 	at com.amazonaws.services.s3.AmazonS3Client.invoke(AmazonS3Client.java:5211)
[2025-05-20T20:48:04.476+0000] {subprocess.py:93} INFO - 	at com.amazonaws.services.s3.AmazonS3Client.invoke(AmazonS3Client.java:5173)
[2025-05-20T20:48:04.477+0000] {subprocess.py:93} INFO - 	at com.amazonaws.services.s3.AmazonS3Client.invoke(AmazonS3Client.java:5167)
[2025-05-20T20:48:04.484+0000] {subprocess.py:93} INFO - 	at com.amazonaws.services.s3.AmazonS3Client.listObjectsV2(AmazonS3Client.java:963)
[2025-05-20T20:48:04.510+0000] {subprocess.py:93} INFO - 	at org.apache.hadoop.fs.s3a.S3AFileSystem.lambda$listObjects$7(S3AFileSystem.java:2116)
[2025-05-20T20:48:04.512+0000] {subprocess.py:93} INFO - 	at org.apache.hadoop.fs.statistics.impl.IOStatisticsBinding.lambda$trackDurationOfOperation$5(IOStatisticsBinding.java:499)
[2025-05-20T20:48:04.532+0000] {subprocess.py:93} INFO - 	at org.apache.hadoop.fs.s3a.Invoker.retryUntranslated(Invoker.java:412)
[2025-05-20T20:48:04.533+0000] {subprocess.py:93} INFO - 	at org.apache.hadoop.fs.s3a.Invoker.retryUntranslated(Invoker.java:375)
[2025-05-20T20:48:04.536+0000] {subprocess.py:93} INFO - 	at org.apache.hadoop.fs.s3a.S3AFileSystem.listObjects(S3AFileSystem.java:2107)
[2025-05-20T20:48:04.554+0000] {subprocess.py:93} INFO - 	at org.apache.hadoop.fs.s3a.S3AFileSystem.s3GetFileStatus(S3AFileSystem.java:3322)
[2025-05-20T20:48:04.556+0000] {subprocess.py:93} INFO - 	... 21 more
[2025-05-20T20:48:04.557+0000] {subprocess.py:93} INFO - Caused by: com.amazonaws.SdkClientException: Unable to load AWS credentials from environment variables (AWS_ACCESS_KEY_ID (or AWS_ACCESS_KEY) and AWS_SECRET_KEY (or AWS_SECRET_ACCESS_KEY))
[2025-05-20T20:48:04.557+0000] {subprocess.py:93} INFO - 	at com.amazonaws.auth.EnvironmentVariableCredentialsProvider.getCredentials(EnvironmentVariableCredentialsProvider.java:50)
[2025-05-20T20:48:04.558+0000] {subprocess.py:93} INFO - 	at org.apache.hadoop.fs.s3a.AWSCredentialProviderList.getCredentials(AWSCredentialProviderList.java:177)
[2025-05-20T20:48:04.559+0000] {subprocess.py:93} INFO - 	... 43 more
[2025-05-20T20:48:05.971+0000] {subprocess.py:93} INFO - 25/05/20 20:48:05 INFO CoarseGrainedSchedulerBackend$DriverEndpoint: Registered executor NettyRpcEndpointRef(spark-client://Executor) (172.18.0.7:47828) with ID 0,  ResourceProfileId 0
[2025-05-20T20:48:06.429+0000] {subprocess.py:93} INFO - 25/05/20 20:48:06 INFO BlockManagerMasterEndpoint: Registering block manager 172.18.0.7:39681 with 366.3 MiB RAM, BlockManagerId(0, 172.18.0.7, 39681, None)
[2025-05-20T20:48:06.790+0000] {subprocess.py:93} INFO - 2025-05-20 20:48:06,789 - ERROR - ETL process failed: An error occurred while calling o31.parquet.
[2025-05-20T20:48:06.794+0000] {subprocess.py:93} INFO - : java.nio.file.AccessDeniedException: s3a://nyc-taxi-data/yellow_taxi: org.apache.hadoop.fs.s3a.auth.NoAuthWithAWSException: No AWS Credentials provided by TemporaryAWSCredentialsProvider SimpleAWSCredentialsProvider EnvironmentVariableCredentialsProvider IAMInstanceCredentialsProvider : com.amazonaws.SdkClientException: Unable to load AWS credentials from environment variables (AWS_ACCESS_KEY_ID (or AWS_ACCESS_KEY) and AWS_SECRET_KEY (or AWS_SECRET_ACCESS_KEY))
[2025-05-20T20:48:06.797+0000] {subprocess.py:93} INFO - 	at org.apache.hadoop.fs.s3a.S3AUtils.translateException(S3AUtils.java:206)
[2025-05-20T20:48:06.802+0000] {subprocess.py:93} INFO - 	at org.apache.hadoop.fs.s3a.Invoker.once(Invoker.java:117)
[2025-05-20T20:48:06.823+0000] {subprocess.py:93} INFO - 	at org.apache.hadoop.fs.s3a.S3AFileSystem.listStatus(S3AFileSystem.java:2810)
[2025-05-20T20:48:06.856+0000] {subprocess.py:93} INFO - 	at org.apache.hadoop.fs.Globber.listStatus(Globber.java:128)
[2025-05-20T20:48:06.857+0000] {subprocess.py:93} INFO - 	at org.apache.hadoop.fs.Globber.doGlob(Globber.java:291)
[2025-05-20T20:48:06.878+0000] {subprocess.py:93} INFO - 	at org.apache.hadoop.fs.Globber.glob(Globber.java:202)
[2025-05-20T20:48:06.881+0000] {subprocess.py:93} INFO - 	at org.apache.hadoop.fs.s3a.S3AFileSystem.globStatus(S3AFileSystem.java:4253)
[2025-05-20T20:48:06.893+0000] {subprocess.py:93} INFO - 	at org.apache.hadoop.fs.s3a.S3AFileSystem.globStatus(S3AFileSystem.java:4233)
[2025-05-20T20:48:06.910+0000] {subprocess.py:93} INFO - 	at org.apache.spark.deploy.SparkHadoopUtil.globPath(SparkHadoopUtil.scala:253)
[2025-05-20T20:48:06.912+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.DataSource$.$anonfun$checkAndGlobPathIfNecessary$3(DataSource.scala:765)
[2025-05-20T20:48:06.914+0000] {subprocess.py:93} INFO - 	at org.apache.spark.util.ThreadUtils$.$anonfun$parmap$2(ThreadUtils.scala:372)
[2025-05-20T20:48:06.927+0000] {subprocess.py:93} INFO - 	at scala.concurrent.Future$.$anonfun$apply$1(Future.scala:659)
[2025-05-20T20:48:06.930+0000] {subprocess.py:93} INFO - 	at scala.util.Success.$anonfun$map$1(Try.scala:255)
[2025-05-20T20:48:06.949+0000] {subprocess.py:93} INFO - 	at scala.util.Success.map(Try.scala:213)
[2025-05-20T20:48:06.951+0000] {subprocess.py:93} INFO - 	at scala.concurrent.Future.$anonfun$map$1(Future.scala:292)
[2025-05-20T20:48:06.952+0000] {subprocess.py:93} INFO - 	at scala.concurrent.impl.Promise.liftedTree1$1(Promise.scala:33)
[2025-05-20T20:48:06.953+0000] {subprocess.py:93} INFO - 	at scala.concurrent.impl.Promise.$anonfun$transform$1(Promise.scala:33)
[2025-05-20T20:48:06.964+0000] {subprocess.py:93} INFO - 	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:64)
[2025-05-20T20:48:06.970+0000] {subprocess.py:93} INFO - 	at java.base/java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1426)
[2025-05-20T20:48:06.992+0000] {subprocess.py:93} INFO - 	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
[2025-05-20T20:48:07.001+0000] {subprocess.py:93} INFO - 	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
[2025-05-20T20:48:07.026+0000] {subprocess.py:93} INFO - 	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
[2025-05-20T20:48:07.035+0000] {subprocess.py:93} INFO - 	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
[2025-05-20T20:48:07.036+0000] {subprocess.py:93} INFO - 	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)
[2025-05-20T20:48:07.037+0000] {subprocess.py:93} INFO - Caused by: org.apache.hadoop.fs.s3a.auth.NoAuthWithAWSException: No AWS Credentials provided by TemporaryAWSCredentialsProvider SimpleAWSCredentialsProvider EnvironmentVariableCredentialsProvider IAMInstanceCredentialsProvider : com.amazonaws.SdkClientException: Unable to load AWS credentials from environment variables (AWS_ACCESS_KEY_ID (or AWS_ACCESS_KEY) and AWS_SECRET_KEY (or AWS_SECRET_ACCESS_KEY))
[2025-05-20T20:48:07.038+0000] {subprocess.py:93} INFO - 	at org.apache.hadoop.fs.s3a.AWSCredentialProviderList.getCredentials(AWSCredentialProviderList.java:216)
[2025-05-20T20:48:07.039+0000] {subprocess.py:93} INFO - 	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.getCredentialsFromContext(AmazonHttpClient.java:1257)
[2025-05-20T20:48:07.056+0000] {subprocess.py:93} INFO - 	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.runBeforeRequestHandlers(AmazonHttpClient.java:833)
[2025-05-20T20:48:07.067+0000] {subprocess.py:93} INFO - 	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.doExecute(AmazonHttpClient.java:783)
[2025-05-20T20:48:07.084+0000] {subprocess.py:93} INFO - 	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeWithTimer(AmazonHttpClient.java:770)
[2025-05-20T20:48:07.104+0000] {subprocess.py:93} INFO - 	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.execute(AmazonHttpClient.java:744)
[2025-05-20T20:48:07.116+0000] {subprocess.py:93} INFO - 	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.access$500(AmazonHttpClient.java:704)
[2025-05-20T20:48:07.120+0000] {subprocess.py:93} INFO - 	at com.amazonaws.http.AmazonHttpClient$RequestExecutionBuilderImpl.execute(AmazonHttpClient.java:686)
[2025-05-20T20:48:07.128+0000] {subprocess.py:93} INFO - 	at com.amazonaws.http.AmazonHttpClient.execute(AmazonHttpClient.java:550)
[2025-05-20T20:48:07.130+0000] {subprocess.py:93} INFO - 	at com.amazonaws.http.AmazonHttpClient.execute(AmazonHttpClient.java:530)
[2025-05-20T20:48:07.131+0000] {subprocess.py:93} INFO - 	at com.amazonaws.services.s3.AmazonS3Client.invoke(AmazonS3Client.java:5227)
[2025-05-20T20:48:07.133+0000] {subprocess.py:93} INFO - 	at com.amazonaws.services.s3.AmazonS3Client.getBucketRegionViaHeadRequest(AmazonS3Client.java:6189)
[2025-05-20T20:48:07.151+0000] {subprocess.py:93} INFO - 	at com.amazonaws.services.s3.AmazonS3Client.fetchRegionFromCache(AmazonS3Client.java:6162)
[2025-05-20T20:48:07.157+0000] {subprocess.py:93} INFO - 	at com.amazonaws.services.s3.AmazonS3Client.invoke(AmazonS3Client.java:5211)
[2025-05-20T20:48:07.166+0000] {subprocess.py:93} INFO - 	at com.amazonaws.services.s3.AmazonS3Client.invoke(AmazonS3Client.java:5173)
[2025-05-20T20:48:07.169+0000] {subprocess.py:93} INFO - 	at com.amazonaws.services.s3.AmazonS3Client.invoke(AmazonS3Client.java:5167)
[2025-05-20T20:48:07.171+0000] {subprocess.py:93} INFO - 	at com.amazonaws.services.s3.AmazonS3Client.listObjectsV2(AmazonS3Client.java:963)
[2025-05-20T20:48:07.172+0000] {subprocess.py:93} INFO - 	at org.apache.hadoop.fs.s3a.S3AFileSystem.lambda$listObjects$7(S3AFileSystem.java:2116)
[2025-05-20T20:48:07.184+0000] {subprocess.py:93} INFO - 	at org.apache.hadoop.fs.statistics.impl.IOStatisticsBinding.lambda$trackDurationOfOperation$5(IOStatisticsBinding.java:499)
[2025-05-20T20:48:07.202+0000] {subprocess.py:93} INFO - 	at org.apache.hadoop.fs.s3a.Invoker.retryUntranslated(Invoker.java:412)
[2025-05-20T20:48:07.222+0000] {subprocess.py:93} INFO - 	at org.apache.hadoop.fs.s3a.Invoker.retryUntranslated(Invoker.java:375)
[2025-05-20T20:48:07.235+0000] {subprocess.py:93} INFO - 	at org.apache.hadoop.fs.s3a.S3AFileSystem.listObjects(S3AFileSystem.java:2107)
[2025-05-20T20:48:07.251+0000] {subprocess.py:93} INFO - 	at org.apache.hadoop.fs.s3a.S3AFileSystem$ListingOperationCallbacksImpl.lambda$listObjectsAsync$0(S3AFileSystem.java:1750)
[2025-05-20T20:48:07.254+0000] {subprocess.py:93} INFO - 	at org.apache.hadoop.fs.s3a.impl.CallableSupplier.get(CallableSupplier.java:62)
[2025-05-20T20:48:07.266+0000] {subprocess.py:93} INFO - 	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1700)
[2025-05-20T20:48:07.270+0000] {subprocess.py:93} INFO - 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
[2025-05-20T20:48:07.276+0000] {subprocess.py:93} INFO - 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
[2025-05-20T20:48:07.287+0000] {subprocess.py:93} INFO - 	at java.base/java.lang.Thread.run(Thread.java:829)
[2025-05-20T20:48:07.291+0000] {subprocess.py:93} INFO - Caused by: com.amazonaws.SdkClientException: Unable to load AWS credentials from environment variables (AWS_ACCESS_KEY_ID (or AWS_ACCESS_KEY) and AWS_SECRET_KEY (or AWS_SECRET_ACCESS_KEY))
[2025-05-20T20:48:07.293+0000] {subprocess.py:93} INFO - 	at com.amazonaws.auth.EnvironmentVariableCredentialsProvider.getCredentials(EnvironmentVariableCredentialsProvider.java:50)
[2025-05-20T20:48:07.295+0000] {subprocess.py:93} INFO - 	at org.apache.hadoop.fs.s3a.AWSCredentialProviderList.getCredentials(AWSCredentialProviderList.java:177)
[2025-05-20T20:48:07.296+0000] {subprocess.py:93} INFO - 	... 27 more
[2025-05-20T20:48:07.298+0000] {subprocess.py:93} INFO - 
[2025-05-20T20:48:07.309+0000] {subprocess.py:93} INFO - 25/05/20 20:48:06 INFO SparkUI: Stopped Spark web UI at http://54a986196d48:4040
[2025-05-20T20:48:07.314+0000] {subprocess.py:93} INFO - 25/05/20 20:48:06 INFO StandaloneSchedulerBackend: Shutting down all executors
[2025-05-20T20:48:07.316+0000] {subprocess.py:93} INFO - 25/05/20 20:48:07 INFO CoarseGrainedSchedulerBackend$DriverEndpoint: Asking each executor to shut down
[2025-05-20T20:48:07.340+0000] {subprocess.py:93} INFO - 25/05/20 20:48:07 INFO MapOutputTrackerMasterEndpoint: MapOutputTrackerMasterEndpoint stopped!
[2025-05-20T20:48:07.358+0000] {subprocess.py:93} INFO - 25/05/20 20:48:07 INFO MemoryStore: MemoryStore cleared
[2025-05-20T20:48:07.373+0000] {subprocess.py:93} INFO - 25/05/20 20:48:07 INFO BlockManager: BlockManager stopped
[2025-05-20T20:48:07.432+0000] {subprocess.py:93} INFO - 25/05/20 20:48:07 INFO BlockManagerMaster: BlockManagerMaster stopped
[2025-05-20T20:48:07.433+0000] {subprocess.py:93} INFO - 25/05/20 20:48:07 INFO OutputCommitCoordinator$OutputCommitCoordinatorEndpoint: OutputCommitCoordinator stopped!
[2025-05-20T20:48:07.559+0000] {subprocess.py:93} INFO - 25/05/20 20:48:07 ERROR TransportRequestHandler: Error sending result StreamResponse[streamId=/files/com.amazonaws_aws-java-sdk-bundle-1.11.901.jar,byteCount=192321266,body=FileSegmentManagedBuffer[file=/home/airflow/.ivy2/jars/com.amazonaws_aws-java-sdk-bundle-1.11.901.jar,offset=0,length=192321266]] to /172.18.0.7:47838; closing connection
[2025-05-20T20:48:07.562+0000] {subprocess.py:93} INFO - io.netty.channel.StacklessClosedChannelException
[2025-05-20T20:48:07.564+0000] {subprocess.py:93} INFO - 	at io.netty.channel.AbstractChannel.close(ChannelPromise)(Unknown Source)
[2025-05-20T20:48:07.639+0000] {subprocess.py:93} INFO - 25/05/20 20:48:07 INFO SparkContext: Successfully stopped SparkContext
[2025-05-20T20:48:08.071+0000] {subprocess.py:93} INFO - 25/05/20 20:48:08 INFO ShutdownHookManager: Shutdown hook called
[2025-05-20T20:48:08.073+0000] {subprocess.py:93} INFO - 25/05/20 20:48:08 INFO ShutdownHookManager: Deleting directory /tmp/spark-5112f341-e159-431d-98b9-321d96296919
[2025-05-20T20:48:08.104+0000] {subprocess.py:93} INFO - 25/05/20 20:48:08 INFO ShutdownHookManager: Deleting directory /tmp/spark-d60f1fe2-342e-4414-a5fa-8338cb3959b9
[2025-05-20T20:48:08.114+0000] {subprocess.py:93} INFO - 25/05/20 20:48:08 INFO ShutdownHookManager: Deleting directory /tmp/spark-d60f1fe2-342e-4414-a5fa-8338cb3959b9/pyspark-b84fe6e2-88f8-4d8b-a392-9607e483445c
[2025-05-20T20:48:08.137+0000] {subprocess.py:93} INFO - 25/05/20 20:48:08 INFO MetricsSystemImpl: Stopping s3a-file-system metrics system...
[2025-05-20T20:48:08.143+0000] {subprocess.py:93} INFO - 25/05/20 20:48:08 INFO MetricsSystemImpl: s3a-file-system metrics system stopped.
[2025-05-20T20:48:08.145+0000] {subprocess.py:93} INFO - 25/05/20 20:48:08 INFO MetricsSystemImpl: s3a-file-system metrics system shutdown complete.
[2025-05-20T20:48:08.329+0000] {subprocess.py:97} INFO - Command exited with return code 0
[2025-05-20T20:48:08.418+0000] {taskinstance.py:1398} INFO - Marking task as SUCCESS. dag_id=yellow_taxi_batch_pipeline, task_id=transform_taxi_data, execution_date=20250520T204538, start_date=20250520T204720, end_date=20250520T204808
[2025-05-20T20:48:08.512+0000] {local_task_job_runner.py:228} INFO - Task exited with return code 0
[2025-05-20T20:48:08.591+0000] {taskinstance.py:2776} INFO - 0 downstream tasks scheduled from follow-on schedule check
